src/Glpi/Controller/Form/TagListController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller\Form;
+
+use Glpi\Controller\Controller;
+use Glpi\Form\Form;
+use Glpi\Form\Tag\FormTagsManager;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Attribute\Route;
+
+final class TagListController implements Controller
+{
+    #[Route(
+        "/ajax/Form/TagList",
```

Comment:
IMHO, we should try to have URLs that matches the classes names. For instance, the URL should probably be only  `/Form/TagList`.

- If you want the `ajax` keyword to appear in the URL, then you should rename the class to `AjaxTagListController` and change the URL to `/Form/AjaxTagList`;
- If you want the `ajax` keyword to appear in the controller class name but not in the URL, then we could maybe use suffixes for controllers classes, like `TagListAjaxController`, `FooHtmlController`, `BarPageController`, ..., but then we will have to document the prefixes that we should use and the corresponding use cases.

Reply:
Nvm, discussed IRL and there should be a way to avoid this.